chronos/timer.nim

```diff
@@ -8,45 +8,397 @@
+  result.value = cast[int64](v)
+
+func microseconds*(v: SomeInteger): Duration {.inline.} =
+  ## Initialize Duration with microseconds value ``v``.
+  result.value = cast[int64](v) * Microsecond.value
+
+func milliseconds*(v: SomeInteger): Duration {.inline.} =
+  ## Initialize Duration with milliseconds value ``v``.
+  result.value = cast[int64](v) * Millisecond.value
+
+func seconds*(v: SomeInteger): Duration {.inline.} =
+  ## Initialize Duration with seconds value ``v``.
+  result.value = cast[int64](v) * Second.value
+
+func minutes*(v: SomeInteger): Duration {.inline.} =
+  ## Initialize Duration with minutes value ``v``.
+  result.value = cast[int64](v) * Minute.value
+
+func hours*(v: SomeInteger): Duration {.inline.} =
+  ## Initialize Duration with hours value ``v``.
+  result.value = cast[int64](v) * Hour.value
+
+func days*(v: SomeInteger): Duration {.inline.} =
+  ## Initialize Duration with days value ``v``.
+  result.value = cast[int64](v) * Day.value
+
+func weeks*(v: SomeInteger): Duration {.inline.} =
+  ## Initialize Duration with weeks value ``v``.
+  result.value = cast[int64](v) * Week.value
+
+func nanoseconds*(v: Duration): Duration {.inline.} =
+  ## Round Duration ``v`` to nanoseconds.
+  result = v
+
+func microseconds*(v: Duration): Duration {.inline.} =
+  ## Round Duration ``v`` to microseconds.
+  result = (v div Microsecond) * Microsecond
+
+func milliseconds*(v: Duration): Duration {.inline.} =
+  ## Round Duration ``v`` to milliseconds.
+  result = (v div Millisecond) * Millisecond
+
+func seconds*(v: Duration): Duration {.inline.} =
+  ## Round Duration ``v`` to seconds.
+  result = (v div Second) * Second
+
+func minutes*(v: Duration): Duration {.inline.} =
+  ## Round Duration ``v`` to minutes.
+  result = (v div Minute) * Minute
+
+func hours*(v: Duration): Duration {.inline.} =
+  ## Round Duration ``v`` to hours.
+  result = (v div Hour) * Hour
+
+func days*(v: Duration): Duration {.inline.} =
+  ## Round Duration ``v`` to days.
+  result = (v div Day) * Day
+
+func weeks*(v: Duration): Duration {.inline.} =
+  ## Round Duration ``v`` to weeks.
+  result = (v div Week) * Week
+
+func nanos*[T: SomeInteger|Duration](v: T): Duration {.inline.} =
+  nanoseconds(v)
+
+func micros*[T: SomeInteger|Duration](v: T): Duration {.inline.} =
+  microseconds(v)
+
+func millis*[T: SomeInteger|Duration](v: T): Duration {.inline.} =
+  milliseconds(v)
+
+func secs*[T: SomeInteger|Duration](v: T): Duration {.inline.} =
+  seconds(v)
+
+func toInt*(a: Duration, unit: Duration): int64 {.inline.} =
+  ## Round Duration ``a`` to specific Duration ``unit`` and returns value
+  ## as integer.
+  result = (a div unit).value
+
+func `$`*(a: Duration): string {.inline.} =
+  ## Returns string representation of Duration ``a`` as nanoseconds value.
+  result = $(a.value)
+
+func `$`*(a: Moment): string {.inline.} =
+  ## Returns string representation of Moment ``a`` as nanoseconds value.
+  result = $(a.value)
```

Comment:
hm, it's questionable that this should be printed ever.. given that it's pretty much just a random value

Reply:
Everything should have printed version.

Reply:
I just do not want to see `rust` biggest problem (unable to print everything) in Nim.